src/status_im/contexts/shell/share/wallet/view.cljs

```diff
@@ -74,7 +74,7 @@
             :emoji               (:emoji account)
             :on-multichain-press #(reset! wallet-type :multichain)
             :on-legacy-press     #(reset! wallet-type :legacy)
-            :on-settings-press   #(open-preferences @selected-networks)}]]]))))
+            :on-settings-press   #(open-preferences selected-networks)}]]]))))
```

Comment:
Hmm, not particularly a fan of passing ratoms through to functions. Is this something we can avoid? 
It seems better if a component can just take the value as a prop instead. Do we have a better approach for situations like this?
cc @ilmotta, @ulisesmac, @Parveshdhull

Reply:
Hi, In this particular scenario, we are modifying the state of current component based on what actions are performed on the next screen.
If we don't want to directly pass this ratom, we can instead pass the function that does the modification in the end ([on-save](https://github.com/status-im/status-mobile/blob/9362d12bbddea497f4efc943f0510d3194e1d09d/src/status_im/contexts/shell/share/wallet/view.cljs#L43))

Reply:
These functions are constant (`:on-settings-press`, `:on-legacy-press` and `on-multichain-press`) in terms of they always do the same.

Currently, each time the atoms `selected-networks` or `wallet-type` change, the component will regenerate the involved functions, but there's no need of it.

We can define them in the form-2 initializing section:
```clojure
(defn wallet-qr-code-item-internal [props]
  (let [{:keys [account width index]} props
        selected-networks             (reagent/atom [:ethereum :optimism :arbitrum])
        wallet-type                   (reagent/atom :legacy)
        on-legacy-press               #(reset! wallet-type :legacy) ;; Here
        on-settings-press             #(open-preferences @selected-networks) ;; Here
        ...
        ]
    (fn [...]
      ...)))
```

So that we will pass the exact same object (function) to the child (`quo/share-qr-code`), it helps to avoid undesired rerenders.


@J-Son89 I think sometimes it's actually needed to deref an atom inside a callback function, but most of the times we can define that function once instead of on each rerender.